daemon/cmd/daemon_main.go

```diff
@@ -1874,6 +1874,9 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 		log.WithError(err).Warn("Failed to send agent start monitor message")
 	}
 
+	// Watches for node neighbors link updates.
+	d.nodeDiscovery.Manager.StartNodeNeighborLinkUpdater(d.datapath.NodeNeighbors())
```

Comment:
I was not entirely sure this will be true across all use cases as nodeNeighbor refreshes are spread across the code base and wanted to be sure someone answers the phone when updates are requested. 
It seems there was different combinations ie nodeNeighDiscoveryEnabled but also DatapathLbMode.
Also @borkmann updates seems to intro another usecase ie add clustermesh nodes into the mix.